fluent-bundle/tests/types_test.rs

```diff
@@ -154,3 +173,25 @@ fn fluent_number_to_operands() {
         }
     );
 }
+
+#[test]
+fn fluent_number_grouping() {
+    let langid_ars = langid!("ccp");
+    let mut bundle: FluentBundle<FluentResource> = FluentBundle::new(vec![langid_ars]);
+    #[allow(deprecated)]
+    bundle.set_icu_data_provider(Box::new(icu_testdata::any_no_fallback()));
```

Comment:
This is deprecated, and it tells us to use `compiled_data` instead.

From a first look, you probably need a new type that implements [`DynamicDataProvider`](https://docs.rs/icu_provider/1.4.0/icu_provider/trait.DynamicDataProvider.html) using [`icu_decimal::provider::Baked`](https://docs.rs/icu_decimal/latest/icu_decimal/provider/struct.Baked.html) by asserting the data key [passed in](https://docs.rs/icu_provider/1.4.0/icu_provider/trait.DynamicDataProvider.html) is equal to [`DecimalSymbolsV1Marker`](https://docs.rs/icu_decimal/latest/icu_decimal/provider/struct.DecimalSymbolsV1Marker.html#impl-KeyedDataMarker-for-DecimalSymbolsV1Marker).

Then you can use [`as_any_provider`](https://docs.rs/icu_provider/latest/icu_provider/any/trait.AsDynamicDataProviderAnyMarkerWrap.html#tymethod.as_any_provider) to convert that to the provider this needs.

Reply:
Thanks, I've added a baked data provider that can be used both from tests/examples and by users of the library.